bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -606,24 +610,31 @@ def run_epoch( self ):
         self.prometheus_gauges.labels("dividends").set( self.metagraph.dividends[self.uid] )
         self.prometheus_gauges.labels("emission").set( self.metagraph.emission[self.uid] )
 
+        # Iterate epochs.
+        self.epoch += 1
+
     def metagraph_sync(self):
         r""" Syncing metagraph together with other metagraph-size related objects
         """
-        old_hotkeys = self.neuron_hotkeys if self.neuron_hotkeys else self.metagraph.hotkeys
-        self.metagraph.sync()
+        old_hotkeys = self.neuron_hotkeys + [] if self.neuron_hotkeys else self.metagraph.hotkeys
+        self.metagraph.sync(cached=self.config.neuron.metagraph_cached or self.subtensor.network != 'local')
```

Comment:
So if the network isn't local, the metagraph will always be synced using the cached?

Reply:
Yes, that is the current validator setting to always use metagraph cache in any case, and definitely want to use cache if not local because it can take an hour to update from nakamoto otherwise. Local update without cache takes about 10 minutes, and gives a more recent state than IPFS cache which is updated ~20 mins.